src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpImmediate.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Xunit;
+using ProjectUtils = Microsoft.VisualStudio.IntegrationTest.Utilities.Common.ProjectUtils;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpImmediate : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpImmediate(VisualStudioInstanceFactory instanceFactory) : base(instanceFactory)
+        {
+            VisualStudio.SolutionExplorer.CreateSolution(nameof(CSharpInteractive));
+            var testProj = new ProjectUtils.Project("TestProj");
+            VisualStudio.SolutionExplorer.AddProject(testProj, WellKnownProjectTemplates.ConsoleApplication, LanguageNames.CSharp);
+        }
+
+        [Fact]
+        public void DumpLocalVaribleValue()
+        {
+            VisualStudio.Editor.SetText(@"
+class Program
+{
+    static void Main(string[] args)
+    {
+        int n1 = 42;
+        int n2 = 43;
+    }
+}
+");
+
+            VisualStudio.Workspace.WaitForAsyncOperations(FeatureAttribute.Workspace);
+            VisualStudio.Debugger.SetBreakPoint("Program.cs", "}");
+            VisualStudio.Debugger.Go(waitForBreakMode: true);
+            VisualStudio.ImmediateWindow.ShowImmediateWindow(clearAll: true);
+            VisualStudio.SendKeys.Send("?n");
+            VisualStudio.Workspace.WaitForAsyncOperations(FeatureAttribute.CompletionSet);
+            VisualStudio.SendKeys.Send("1", VirtualKey.Enter);
```

Comment:
Given this is typing out the 1, it means a lack of completion would still pass. Should it send a tab instead?

Reply:
good point, let me try it. Is there a way to validate the completion is up in your integration tests without going deep into editor bits?
Btw, Apex can do that, it also fully supports debugger toolwindows :)

Reply:
Found a way to make these tests rely on completion by sending Tab as you suggested.